### PR TITLE
Allow parameterized prepared statements in SHOW STATS

### PR DESCRIPTION
**Pull request by MiguelWeezardo**

### CODE REVIEW

File: core/trino-parser/src/main/java/io/trino/sql/tree/DefaultTraversalVisitor.java [hunk 1]
```diff
@@ -683,6 +683,13 @@ protected Void visitExplain(Explain node, C context)
         return null;
     }
 
+    @Override
+    protected Void visitShowStats(ShowStats node, C context)
```
Comment (kokosing) [inline core/trino-parser/src/main/java/io/trino/sql/tree/DefaultTraversalVisitor.java:687]: Please add unit tests.

Comment (findepi) [reply core/trino-parser/src/main/java/io/trino/sql/tree/DefaultTraversalVisitor.java:687]: eg in `TestParameterExtractor` but we also need an integration tests, perhaps in  `TestShowStats`, although I am not sure the QueryRunner supports PREPARE + EXECUTE, does it?

Comment (MiguelWeezardo) [reply core/trino-parser/src/main/java/io/trino/sql/tree/DefaultTraversalVisitor.java:687]: I can write both unit and integration tests.

**Review by kokosing — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestShowStats.java [hunk 2]
```diff
@@ -92,6 +93,19 @@ public void testShowStats()
                         "   ('comment', 500.0, 5.0, 0.0, null, null, null), " +
                         "   (null, null, null, null, 5.0, null, null)");
 
+        Session session = Session.builder(getSession())
+                .addPreparedStatement("my_query", "SHOW STATS FOR (SELECT * FROM nation_partitioned WHERE regionkey = ?)")
+                .build();
+
+        assertQuery(session,
+                "EXECUTE my_query USING 1",
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestShowStats.java:101]: so cool!

**Pull request merged**